exercises/bowling/BowlingTest.cs

```diff
@@ -1,207 +1,303 @@
-        var game = RollMany(rolls, new BowlingGame());
-        Assert.Equal(18, game.Score());
+        var sut = new BowlingGame();
+        var previousRolls = new int[] {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10, 7, 1};
+        DoRoll(previousRolls, sut);
+        var actual = sut.Score();
+        Assert.Equal(18, actual);
     }
 
     [Fact(Skip = "Remove to run test")]
     public void Rolling_a_spare_with_the_two_roll_bonus_does_not_get_a_bonus_roll()
     {
-        var rolls = new[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10, 7, 3 };
-        var game = RollMany(rolls, new BowlingGame());
-        Assert.Equal(20, game.Score());
+        var sut = new BowlingGame();
+        var previousRolls = new int[] {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10, 7, 3};
+        DoRoll(previousRolls, sut);
+        var actual = sut.Score();
+        Assert.Equal(20, actual);
     }
 
     [Fact(Skip = "Remove to run test")]
     public void Strikes_with_the_two_roll_bonus_do_not_get_bonus_rolls()
     {
-        var rolls = new[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10, 10, 10 };
-        var game = RollMany(rolls, new BowlingGame());
-        Assert.Equal(30, game.Score());
+        var sut = new BowlingGame();
+        var previousRolls = new int[] {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10, 10, 10};
+        DoRoll(previousRolls, sut);
+        var actual = sut.Score();
+        Assert.Equal(30, actual);
     }
 
     [Fact(Skip = "Remove to run test")]
     public void A_strike_with_the_one_roll_bonus_after_a_spare_in_the_last_frame_does_not_get_a_bonus()
     {
-        var rolls = new[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 7, 3, 10 };
-        var game = RollMany(rolls, new BowlingGame());
-        Assert.Equal(20, game.Score());
+        var sut = new BowlingGame();
+        var previousRolls = new int[] {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 7, 3, 10};
+        DoRoll(previousRolls, sut);
+        var actual = sut.Score();
+        Assert.Equal(20, actual);
     }
 
     [Fact(Skip = "Remove to run test")]
     public void All_strikes_is_a_perfect_game()
     {
-        var rolls = new[] { 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10 };
-        var game = RollMany(rolls, new BowlingGame());
-        Assert.Equal(300, game.Score());
+        var sut = new BowlingGame();
+        var previousRolls = new int[] {10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10};
+        DoRoll(previousRolls, sut);
+        var actual = sut.Score();
+        Assert.Equal(300, actual);
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Rolls_cannot_score_negative_points()
+    {
+        var sut = new BowlingGame();
+        var previousRolls = new int[] {};
+        DoRoll(previousRolls, sut);
+        sut.Roll(-1);
+        var actual = sut.Score();
+        Assert.Null(actual);
```

Comment:
I would prefer the methods that have erroneous input (see the [canonical-data](https://github.com/exercism/problem-specifications/blob/master/exercises/bowling/canonical-data.json#L133)) to throw an exception, instead of returning `null`.